Frameworks/CoreGraphics/CGDataProvider.mm

```diff
@@ -156,16 +421,28 @@ void CGDataProviderRelease(CGDataProviderRef data) {
  @Notes
 */
 CGDataProviderRef CGDataProviderRetain(CGDataProviderRef provider) {
-    CFRetain((id)provider);
-
+    CFRetain(provider);
     return provider;
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 CFTypeID CGDataProviderGetTypeID() {
-    UNIMPLEMENTED();
-    return StubReturn();
-}
\ No newline at end of file
+    return __CGDataProvider::GetTypeID();
+}
+
+#pragma endregion // CGDataProvider Functions
+
+#pragma region Internal CGDataProvider Functions
+
+const void* _CGDataProviderGetData(CGDataProviderRef provider) {
+    return provider ? provider->GetData() : nullptr;
+}
+
+size_t _CGDataProviderGetSize(CGDataProviderRef provider) {
+    return provider ? provider->GetSize() : 0;
```

Comment:
> return provider ? provider->GetSize() : 0; [](start = 3, length = 43)

lets not do ternary statements.
It complicates stuff more than it needs to and makes it easier to lose during merges.